dev/com.ibm.ws.security.social/src/com/ibm/ws/security/social/tai/TAISubjectUtils.java

```diff
@@ -183,18 +184,92 @@ void setUserNameFromJwtClaims(SocialLoginConfig config) {
     String getRealm(AttributeToSubject attributeToSubject, SocialLoginConfig config) throws SettingCustomPropertiesException {
         String realm = attributeToSubject.getMappedRealm();
         if (realm == null) {
+            realm = getDefaultRealm(config);
+            //realm = getDefaultRealmFromAuthorizationEndpoint(config);
+        }
+        return realm;
+    }
+
+    private String getDefaultRealm(SocialLoginConfig config) throws SettingCustomPropertiesException {
+        String realm = null;
+        if (isOpenShiftConfig(config) && useAccessTokenFromRequest(config)) {
+            String ep = getUserApiEndpoint(config);
+            if (ep == null) {
+                if (tc.isDebugEnabled()) {
+                    Tr.debug(tc, "User api endpoint [" + config.getUserApi() + "] is either empty or too short to be a valid URL");
+                }
+                Tr.error(tc, "REALM_NOT_FOUND");
+                throw new SettingCustomPropertiesException();
+            } else {
+                realm = ep;
+            }
+        } else {
+            // original behavior before adding the support for kube type
             realm = getDefaultRealmFromAuthorizationEndpoint(config);
         }
         return realm;
+        
+    }
+
+    private String getUserApiEndpoint(SocialLoginConfig config) {
```

Comment:
I'd rename this to something like `getRealmFromUserApiEndpoint()`. I was initially confused because I thought it was supposed to return the actual, full user API endpoint (e.g. `https://kubernetes.default.svc/apis/authentication.k8s.io/v1/tokenreviews`). But really it's actually extracting the host name from the configured `userApi` value.

This also expects the `userApi` value to contain `/apis/authentication.k8s.io/v1/tokenreviews`, and if it doesn't then the `userApi` value is simply returned as-is for the realm name. I don't know if there's a valid `userApi` value that could be used in the proxy flow that doesn't contain that string. If there is, I don't think this implementation will work. That could be a moot point, though.

Reply:
Renaming is done. In the proxy flow, I think we can safely assume the token review end point will have the "tokenreviews".